src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/transport/OpenDistroSecuritySSLRequestHandler.java

```diff
@@ -81,6 +82,11 @@ public final void messageReceived(T request, TransportChannel channel, Task task
             channel.sendResponse(exception);
             throw exception;
         }
+
+        String channelType = channel.getChannelType();
+        if (!channelType.equals("direct") && !channelType.equals("transport")) {
+            channel = getInnerChannel(channel);
```

Comment:
Also wondering if this check should come after the following `if (!"transport".equals(channel.getChannelType()))` check

Reply:
Please include comments about why this check needs to be there.

Reply:
Sure. Will add comments. Can you verify the logic ?

Reply:
Logic seems fine for PA. As an extra measure we can verify that the returned channel from getInnerChannel is of channelType transport.

Reply:
throw a runtime error if its not basic and transport ?

Reply:
I think so, esp since we're expecting the Wrapper classes `getInnerChannel` to return a transport channel.

Reply:
We are getting a transport channel but the issue is with the channelType, custom implementations may pass